lib/cn/kvs_mblk_desc.h

```diff
@@ -3,23 +3,70 @@
  * Copyright (C) 2015-2022 Micron Technology, Inc.  All rights reserved.
  */
 
-#ifndef HSE_KVS_MBLK_DESC_H
-#define HSE_KVS_MBLK_DESC_H
+#ifndef HSE_MBLK_DESC_H
+#define HSE_MBLK_DESC_H
 
 #include <stdint.h>
 
+#include <hse/error/merr.h>
+
+#include <hse_util/page.h>
+
 #include <hse/types.h>
 
-struct mpool_mcache_map;
+/* MTF_MOCK_DECL(mblk_desc) */
+
 struct mpool;
 
+/* Mblock descriptor */
 struct kvs_mblk_desc {
-    void *                   map_base; /* base address of mcache map */
-    struct mpool_mcache_map *map;      /* mcache map */
-    uint32_t                 map_idx;  /* index of mblk in map */
-    enum hse_mclass          mclass;   /* media class */
-    struct mpool *           ds;       /* mpool dataset */
-    uint64_t                 mbid;    /* mblock id */
+    void *map_base;         // memory mapped address of mblock
+    uint64_t mbid;          // mblock id
+    uint16_t alen_pages;    // allocated length of mblock, in 4K pages
+    uint16_t wlen_pages;    // written length of mblock, in 4K pages
+    uint8_t  mclass;        // media class
```

Comment:
can you `static_assert` HSE_MCLASS_MAX <= UINT8_MAX? Seems like a premature optimization to change from using an enum to me.

Reply:
I will add a static assert.

This struct would be 28 bytes if alen_pages, wlen_pages and mclass were left in full width form.  It is 21 bytes as is.  That's saves 478,752 bytes of memory on a 2 TB database.

Reply:
```
static_assert(HSE_MCLASS_MAX <= UINT8_MAX,
    "struct kvs_mblk_desc member mclass too small to hold enum hse_mclass");
```